typed-racket-lib/typed-racket/private/shallow-rewrite.rkt

```diff
@@ -232,27 +232,29 @@
                                    (cond
                                      [(pair? args) (values (car args) (cdr args))]
                                      [(syntax? args)
-                                      (let ([e (syntax-e args)]) (values (car e) (cdr e)))]
+                                      (define e (syntax-e args))
+                                      (values (car e) (cdr e))]
                                      [else
                                       (raise-syntax-error 'shallow-rewrite-top
                                                           "#%plain-lambda formals"
                                                           #'formals
                                                           args)]))
                                  (define check*
-                                   (let ([dom+ (for/fold ([acc '()]) ([dom (in-list dom*)])
-                                                 (if (pair? dom)
-                                                     (cons (cdr dom) acc)
-                                                     acc))])
```

Comment:
Could this be a for/list?

Reply:
We should have a for/reverse, clearly.

Reply:
I'm not all that sure it comes up often enough to be warranted :p I think a lot of the time when the order of a list doesn't matter so using `for/reverse` would be viable, it would also be the case that just using an actually-unordered data structure like a set would be even better.